Floofbot/Services/NicknameAlertService.cs

```diff
@@ -0,0 +1,174 @@
+            var serverConfig = _floofDb.NicknameAlertConfigs.Find(guild.Id);
+
+            if (serverConfig == null || !serverConfig.IsOn || serverConfig.Channel == 0) // not configured/disabled
+            {
+                return;
+            }
+            _channel = await GetChannel(guild, serverConfig.Channel);
+
+            var embed = new EmbedBuilder()
+                .WithDescription($"{removeEmoji.Name}: Remove Nickname\n" +
+                $"{warnEmoji.Name}: Warn\n" +
+                $"{kickEmoji.Name}: Kick\n" +
+                $"{banEmoji.Name}: Ban")
+                .Build();
+
+            var message = await _channel.SendMessageAsync($"{badUser.Mention} ({badUser.Username}#{badUser.Discriminator}) has been " +
+                $"detected with a bad name! What should I do?\n\nNickname: {badUser.Nickname}", false, embed);
+            await message.AddReactionAsync(removeEmoji);
+            await message.AddReactionAsync(kickEmoji);
+            await message.AddReactionAsync(warnEmoji);
+            await message.AddReactionAsync(banEmoji);
+
+            messageDic.Add(message.Id, badUser);
+
+        }
+
+        public async Task ReactionCallback(Cacheable<IUserMessage, ulong> message, ISocketMessageChannel channel, SocketReaction reaction)
+        {
+            var msg = message.Value as IUserMessage;
+
+            if (messageDic.ContainsKey(msg.Id))
+            {
+                SocketGuildUser badUser;
+                messageDic.TryGetValue(msg.Id, out badUser);
+                var moderator = badUser.Guild.GetUser(reaction.UserId);
+
+                if (reaction.Emote.Name.Equals(banEmoji.Name))
+                {
+                    try
+                    {
+                        //sends message to user
+                        EmbedBuilder builder = new EmbedBuilder();
+                        builder.Title = "⚖️ Ban Notification";
+                        builder.Description = $"You have been banned from {badUser.Guild.Name}";
+                        builder.AddField("Reason", "Banned by BOT for an inappropriate name");
+                        builder.Color = Color.DarkOrange;
+                        await badUser.SendMessageAsync("", false, builder.Build());
+
+                        await badUser.Guild.AddBanAsync(badUser, 0, $"{moderator.Username}#{moderator.Discriminator} ({moderator.Id}) -> Inappropriate Name");
+
+                        await channel.SendMessageAsync($"Got it! I banned {badUser.Username}#{badUser.Discriminator}!");
+                    }
+                    catch (Exception ex)
+                    {
+                        await channel.SendMessageAsync("Unable to ban user. Do I have the permissions?");
+                        Log.Error("Unable to ban user for bad name: " + ex);
+                    }
+                    messageDic.Remove(msg.Id);
+                    return;
+                }
+                if (reaction.Emote.Name.Equals(warnEmoji.Name))
```

Comment:
else if instead of if can be used here, and then you don't have to write the return statements
Also, if the Name variables are both strings then just use ==

Reply:
I tried using the == operator but ran into issues, so will leave this for now